### ISSUE TITLE
release process failures

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes-sigs/kind#792]
The following things are wrong with the release process:

- the cross build script used for releasing does not fail if one of the builds fails :warning: 
- the cross build script is not in CI :warning: 
- the kustomize version we are depending on does not build for windows :warning: 
- the release scripts do not perform a clean before building :warning: 

I've uploaded cleanly built binaries for the other platforms, but Windows is still broken, and these need fixing.

TODO:
- [x] fix the cross build script to fail if any build fails
- [x] add the cross build script to presubmit
- [x] upgrade kustomize to a patched (unfortunately unreleased) version that builds for windows
- [x] double check the v0.5.0 binaries and remove windows
- [ ] cut a new release

**Comment by BenTheElder** [comment]
see also: https://github.com/kubernetes-sigs/kind/issues/791, https://github.com/kubernetes-sigs/kustomize/issues/1212, https://github.com/kubernetes-sigs/kustomize/pull/1374

**Comment by sudermanjr** [comment]
Are these release issues also related to this error message I got when using one of the images from the v0.5.0 release?

```
Creating cluster "test-infra" ...
 ✓ Ensuring node image (kindest/node:v1.14.6) 🖼
ERRO[11:55:40] Unable to find image 'kindest/node:v1.14.6@sha256:4d53072ec035d82a70e9d88972d0eb63846706a892438238b6b54a4531c33a9d' locally
ERRO[11:55:40] docker: Error response from daemon: manifest for kindest/node@sha256:4d53072ec035d82a70e9d88972d0eb63846706a892438238b6b54a4531c33a9d not found.
```

If not, I will open another issue.

**Comment by BenTheElder** [comment]
No that is not related.

**Comment by BenTheElder** [comment]
regarding the failure to exit non-zero on build failure: https://ss64.com/bash/wait.html

> If n is not given, all currently active child processes are waited for, and the return status is zero.

:man_facepalming:

**Comment by monopole** [comment]
releasing kustomize for windows at https://github.com/kubernetes-sigs/kustomize/releases is blocked by https://github.com/kubernetes-sigs/kustomize/issues/834.  We need a build and unit test run in the pre-commit.

**Comment by BenTheElder** [comment]
@monopole what about just a cross build? We don't need windows machines or unit tests to ensure build works.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes-sigs/kind/pull/794 should:
- make the cross build included in presubmit
- fail the cross build if any job fails
- upgrade kustomize to HEAD (which is the windows patch commit)

Next we need to cut and tag a patch release.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes-sigs/kind/pull/795 marked the release, pushed the tag, adding notes and uploading binaries

**Comment by BenTheElder** [comment]
https://github.com/kubernetes-sigs/kind/releases/tag/v0.5.1 is tagged now, will follow up to update the v0.5.0 release notes with a notice at the top

**Comment by praseodym** [comment]
> regarding the failure to exit non-zero on build failure: https://ss64.com/bash/wait.html
> 
> > If n is not given, all currently active child processes are waited for, and the return status is zero.
> 
> man_facepalming

But also this:
> Otherwise, the return status is the exit status of the last process or job waited for.
Which means that the merged fix of doing `wait $(jobs -p)` will report the exit code of the slowest build.

Example:
```bash
$ cat test.sh
#!/bin/bash
set -x
(sleep 1; exit 1) &
(sleep 1; exit 2) &
(sleep 2; exit 0) &
wait $(jobs -p)

$ ./test.sh; echo "script exit code: $?"
+ sleep 1
+ sleep 1
+ sleep 2
++ jobs -p
+ wait 23063 23064 23065
+ exit 2
+ exit 1
+ exit 0
script exit code: 0
```

**Comment by BenTheElder** [comment]
Thanks @praseodym, @justinsb also pointed out via another channel that there's a race between the job starting and exiting.

We'll need a more robust fix for that script specifically :disappointed:

**Comment by BenTheElder** [comment]
follow up regarding `wait` https://github.com/kubernetes-sigs/kind/pull/798
